File: BECore/app/model/AcademicRank.js
/// <summary>
/// Model EXTJS AcademicRank.
/// </summary>
Ext.define('QLDT.model.AcademicRank', {
    extend: 'Ext.data.Model',
    idProperty: 'AcademicRankID',
    idType: 'Guid',

    code: 'AcademicRankCode',
    name: 'AcademicRankName',

    fields: [
        { name: "AcademicRankID", type: "string" },
		{ name: "AcademicRankCode", type: "string" },
		{ name: "AcademicRankName", type: "string" },
		{ name: "AcademicRankShortName", type: "string" },
		{ name: "Description", type: "string" },
		{ name: "Inactive", type: "boolean" },
		{ name: "SortOrder", type: "int" },
		{ name: "CreatedDate", type: "string" },
		{ name: "ModifiedDate", type: "string" },
		{ name: "IPAddress", type: "string" },
		{ name: "ModifiedBy", type: "string" },
		{ name: "CreatedBy", type: "string" },
    ]
});

File: MT.Model/Dictionary/AcademicRank.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace MT.Model
{
    #region AcademicRank:EntityBase
    /// <summary>
    /// This object represents the properties and methods of a AcademicRank:EntityBase.
    /// </summary>
    public class AcademicRank : EntityBase
    {
        [Key]
        public Guid AcademicRankID { get; set; }
        public string AcademicRankCode { get; set; }
        public string AcademicRankName { get; set; }
        public string AcademicRankShortName { get; set; }
        public string Description { get; set; }
        public bool Inactive { get; set; }
        public int SortOrder { get; set; }
    }
    #endregion
}

